### ISSUE TITLE
[🐞] Usage of globalAction$ on SSR streaming is not supported

### ISSUE THREAD

**Issue by saeedhbi** [QwikDev/qwik#5458]
### Which component is affected?

Qwik City (routing)

### Describe the bug

I'm building an app with multiple parts that stream server-side rendering (SSR) fragments, each with a component called `globalAction$` inside. 

Here's the hitch: when this global action tries to do its thing, it redirects requests to the main app host instead of sticking to its own turf.

I took a peek and it looks like this hiccup might be because the global action uses `useLocation` to figure out where it's supposed to be, but it's not properly using the fragment's specific `q:base`. Weirdly, other components like events don't seem to have this issue.

I've got a video that shows what's happening, but reproducing this problem on its own is tough. Check it out here: [Video Link](https://drive.google.com/file/d/1LC6fd6t7YMAfg74kus_O3FtULgFQbTXI/view?usp=sharing)

Basically, the video demos two endpoints: one at 5004 for the fragment and another at 5000 for the main app. When the fragment's doing its own SSR thing, the globalAction$ is spot on with its address. But once it's in the main app, that action's lost its way, not connecting correctly to the 5004 address.

I'm not sure if this is a setup thing or a bug. If you've got any clever workarounds or tips to tackle this, I'm all ears. Thanks a bunch for any help!

### Reproduction

-

### System Info

```shell
System:
  Binaries:
    Node: 20.1.0 - ~/.nvm/versions/node/v20.1.0/bin/node
    npm: 9.6.4 - ~/.nvm/versions/node/v20.1.0/bin/npm
    pnpm: 8.5.1 - ~/.nvm/versions/node/v20.1.0/bin/pnpm
  npmPackages:
    @builder.io/qwik-auth: ^0.1.3 => 0.1.3 
    @builder.io/qwik-city: 1.2.6 => 1.2.6 
    @builder.io/qwik: ^1.2.6 => 1.2.6
```

**Comment by saeedhbi** [comment]
Is there any update regarding this issue?

**Comment by saeedhbi** [comment]
Yes. I do have two separated containers served from different servers and I'm streaming one of them in another.

**Comment by saeedhbi** [comment]
OK. I found the problem. The problem is when are calling an event to retrieve data from `server$`, `globalAction$` or ..., the Qwik application tries to resolve the request with `qwik-json` header. The problem is the qwik loader is always trying to resolve the QRL with `?qfunc=${hash}` URL which is not correct since the streamed content should be called by their origin.

**Comment by JerryWu1234** [comment]
@wmertens 
Could you offer more detail how to achieve it

**Comment by gioboa** [comment]
is it still a valid issue?

**Comment by wmertens** [comment]
yes I think this is still a valid issue.
So the server calls happen [here](https://github.com/QwikDev/qwik/blob/17190078f9d0239d10b5f9e21242aca579dc2de8/packages/qwik-city/src/runtime/src/server-functions.ts#L461), and they always go to the origin.
Instead, they should go to q:base, but without the `/build/` or `/build/en/` etc that gets added.
We may need a new property q:server instead

**Comment by JerryWu1234** [comment]
@saeedhbi @wmertens Is there any a minimal repository ? or
or tell me how to reproduce it 
thanks
